serverless/serverless-azure-functions
# feat: Invoke APIM endpoint

## What did you implement:

Invocation for APIM endpoint.



## How did you implement it:

Instead of boolean for `local` invocations, added `InvokeMode` enum for `FUNCTION`, `LOCAL` and `APIM`. Fixed query string to not have the equal sign encoded into `%3D` by using the `querystring` package. Added tests.



## How can we verify it:

Deploy a function app with APIM configured (`apim: true`), and then run `sls invoke apim ...` like you would invoke your function normally. You will see the APIM endpoint logged to the console and the result is returned like normal.



## Todos:

_**Note: Run `npm run test:ci` to run all validation checks on proposed changes**_

       **Validate via `npm run lint`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
       **Validate via `npm test`**

**_Is this ready for review?:_** YES  
**_Is it a breaking change?:_** NO
